includes/documents/class-wcpdf-invoice.php

```diff
@@ -151,13 +145,35 @@ public function get_sequential_number_store() {
 	}
 	
 	public function get_sequential_number_store_name( $date ) {
-		$year       = $date->date_i18n( 'Y' );
-		$store_name = "{$this->slug}_number_{$year}";
-
-		// legacy filter for backward compatibility
-		$store_name = apply_filters( 'wpo_wcpdf_document_sequential_number_store', $store_name, $this );
+		$year            = $date->date_i18n( 'Y' );
+		$store_base_name = apply_filters( 'wpo_wcpdf_document_sequential_number_store', "{$this->slug}_number", $this );
+	
+		// migrate old stores without year suffix
+		$number_store_migrated = get_option( "wpo_wcpdf_number_store_{$store_base_name}_migrated" );
+		if( ! $number_store_migrated ) {
+			$this->maybe_migrate_number_store( $store_base_name );
+		}
+	
+		return apply_filters( "wpo_wcpdf_{$this->slug}_number_store_name", "{$store_base_name}_{$year}", $this );
+	}
 
-		return apply_filters( "wpo_wcpdf_{$this->slug}_number_store_name", $store_name, $this );
+	private function maybe_migrate_number_store( $store_base_name ) {
+		global $wpdb;
+		$now            = new \WC_DateTime( "@".time(), new \DateTimeZone( 'UTC' ) );
+		$year           = $now->date_i18n( 'Y' );
+		$new_table_name = "{$store_base_name}_{$year}";
+
+		if( $wpdb->get_var( "SHOW TABLES LIKE '{$store_base_name}'") == $store_base_name ) {
+			$query = $wpdb->query( "ALTER TABLE {$store_base_name} RENAME {$new_table_name} ");
+			if( $query ) {
+				update_option( "wpo_wcpdf_number_store_{$store_base_name}_migrated", 1 );
+				wcpdf_log_error( sprintf( __( 'Number store migration from %s (old table) to %s (new table) run successfully!', 'woocommerce-pdf-invoices-packing-slips' ), $store_base_name, $new_table_name ), 'info' );
+			} else {
+				wcpdf_log_error( sprintf( __( 'An error occurred while trying to migrate the number store %s: %s', 'woocommerce-pdf-invoices-packing-slips' ), $store_base_name, $wpdb->last_error ) );
+			}
+		} else {
+			wcpdf_log_error( sprintf( __( 'Legacy number store table %s not found: %s', 'woocommerce-pdf-invoices-packing-slips' ), $store_base_name, $wpdb->last_error ), 'warning' );
```

Comment:
this would keep triggering errors for each `Number_Store` instance for sites that never had the legacy store in the first place